x/valset-pref/validator_set.go

```diff
@@ -229,43 +223,46 @@ func (k Keeper) UndelegateFromValidatorSet(ctx sdk.Context, delegatorAddr string
 	return nil
 }
 
+// getValsetRatios returns the valRatio array calculated based on the given delegator, valset prefs, and undelegating amount.
+// Errors when given delegator does not have delegation towards all of the validators given in the valsetPrefs
 func (k Keeper) getValsetRatios(ctx sdk.Context, delegator sdk.AccAddress,
 	prefs []types.ValidatorPreference, undelegateAmt osmomath.Int) ([]ValRatio, map[string]stakingtypes.Validator, osmomath.Dec, error) {
 	// total amount user has delegated
 	totalDelegatedAmt := osmomath.ZeroDec()
-	var valSetRatio []ValRatio
+	var valSetRatios []ValRatio
 	validators := map[string]stakingtypes.Validator{}
 
 	for _, val := range prefs {
 		amountToUnDelegate := val.Weight.MulInt(undelegateAmt).TruncateInt()
 		valAddr, validator, err := k.getValAddrAndVal(ctx, val.ValOperAddress)
 		if err != nil {
-			return valSetRatio, validators, totalDelegatedAmt, err
+			return []ValRatio{}, map[string]stakingtypes.Validator{}, sdk.ZeroDec(), err
```

Comment:
question, does []ValRatio{} not implement nil?

Reply:
Good catch! It does